core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -215,6 +215,18 @@ class TaskMetrics extends Serializable {
     inputMetrics.foreach(_.updateBytesRead())
   }
 
+  override def equals(other: Any): Boolean = other match {
```

Comment:
yeah, that would be preferable actually. In the future if we add more fields to `TaskMetrics` we might forget to add it to `equals`